### ISSUE TITLE
HasQueryFilter not working from Entity Framework Core 3.0 Preview 6

### ISSUE THREAD

**Issue by tallichet** [dotnet/efcore#16139]
I have updated a project (blazor server side using EF Core 3) from preview 5 to preview 6. Since then,  my QueryFilters are not working any more.
My database context are registered as transcient.

The code is has follows:

```
private void SetQueryFilter(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Contact>().HasQueryFilter(c => !c.Deleted && c.TenantId == _tenantId);
}
```

The `SetQueryFilter` is called from `OnModelCreating`. `_tenantId` is set in the ctor.

The methods also seems to be called only once.

**Comment by tallichet** [comment]
And yes, my code was working correctly using Entity Framework Core 3.0 Preview 5

**Comment by smitpatel** [comment]
Duplicate of #15264

**Comment by ajcvickers** [comment]
@tallichet First, many thanks for trying preview 6. We really appreciate people trying the previews and filing issues on what they find.

If you haven't seen it already, there is some good information in the [preview announcement post](https://devblogs.microsoft.com/dotnet/announcing-entity-framework-core-3-0-preview-6-and-entity-framework-6-3-preview-6/) about the changes that are happening to LINQ queries for EF Core 3.0. Preview 6 is the first release containing these changes, which means we are both expecting things to be broken, but at the same time very grateful for everyone who tries the release and generated feedback.

Support for query filters was not completed for preview 6. Closing this as a duplicate of #15264
